Src/Newtonsoft.Json/JsonConvert.cs

```diff
@@ -962,9 +962,15 @@ public static void PopulateObject(string value, object target, JsonSerializerSet
             {
                 jsonSerializer.Populate(jsonReader, target);
 
-                if (jsonReader.Read() && jsonReader.TokenType != JsonToken.Comment)
+                if (settings != null && settings.CheckAdditionalContent)
                 {
-                    throw new JsonSerializationException("Additional text found in JSON string after finishing deserializing object.");
+                    while (jsonReader.Read())
+                    {
+                        if (jsonReader.TokenType != JsonToken.Comment)
+                        {
+                            throw new JsonSerializationException("Additional text found in JSON string after finishing deserializing object.");
```

Comment:
Change this exception to use `JsonSerializationException.Create(reader, "blah blah")`